app/scripts/metamask-controller.js

```diff
@@ -116,16 +116,34 @@ export default class MetamaskController extends EventEmitter {
     })
 
     // This depends on preferences controller state
-    this.trackSegmentEvent = getTrackSegmentEvent(
+    this.trackSegmentEvent = getTrackMetaMetricsEvent(
       this.platform.getVersion(),
-      () => this.preferencesController.getParticipateInMetaMetrics(),
       () => {
+        const participateInMetaMetrics = this.preferencesController.getParticipateInMetaMetrics()
         const {
-          currentLocale,
           metaMetricsId,
+          currentLocale,
```

Comment:
lol noooo. I went through a few iterations of api design before landing on this and at one point these two things were in separate functions 👀 lol. I, too, prefer order over chaos and I will fix this to avoid complicating the git history.

Reply:
😂